packages/react/src/components/Modal/Modal.js

```diff
@@ -303,12 +312,28 @@ export default class Modal extends Component {
       </button>
     );
 
+    const getAriaLabelledBy = (() => {
+      const ariaLabelledBy = [];
+      if (modalLabel) {
+        ariaLabelledBy.push(
+          `${prefix}--modal-header__label`,
+          `${prefix}--modal-header__heading`
+        );
+      }
+      return ariaLabelledBy.length ? ariaLabelledBy.join(' ') : null;
+    })();
+
     const modalBody = (
       <div
         ref={this.innerModal}
         role="dialog"
         className={`${prefix}--modal-container`}
-        aria-label={modalAriaLabel}
+        aria-label={
+          modalLabel
```

Comment:
Good stuff 👍 I'll have to remember this technique for if this if/else labeling pattern comes up again in the future.